Example with collecting timestamp of the metrics

andreyvelich opened the pull request:
Fixes: https://github.com/kubeflow/katib/issues/944.
Hold until I change docker hub repository.
/hold

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/katib/970)
<!-- Reviewable:end -->

hougangliu commented on examples/v1alpha3/metrics-with-timestamp-example.yaml:
```diff
@@ -0,0 +1,53 @@
+              spec:
+                containers:
+                - name: {{.Trial}}
+                  image: docker.io/andreyvelichkevich/timestamp-metric
```
I had added you into kubeflowkatib, you can now push kubeflowkatib/timestamp-metric now
BTW, I think you can rename the image name to `kubeflowkatib/mxnet-mnist-example` to make all example has timestamp.

johnugeorge replied:
I suggest to rename folder and file name also to `metrics-with-timestamp` to `mxnet-mnist-example`. There is no need  to highlight the term`with-timestamp`

andreyvelich replied:
This is not a `mxnet-mnist-example`. 
I took code of pytorch-mnist from here: https://github.com/kubeflow/katib/blob/master/examples/v1alpha3/file-metrics-collector/mnist.py. 

Should I change example to Mxnet in that case? Mxnet example is that one: https://github.com/apache/incubator-mxnet/blob/master/example/image-classification/train_mnist.py.
Or we can name this example `kubeflowkatib/pytorch-mnist-example`?

johnugeorge replied:
I missed it. IIRC @richardsliu created this image to show mxnet support.

andreyvelich replied:
@johnugeorge So what can we do with this example?

johnugeorge replied:
@richardsliu Can you point out at the Dockerfile used?

vpavlin commented on examples/v1alpha3/metrics-with-timestamp/Dockerfile:
```diff
@@ -0,0 +1,6 @@
+FROM pytorch/pytorch:1.0-cuda10.0-cudnn7-runtime
+
+WORKDIR /var
```
I don't believe /var is appropriate location for the script. If you check FSH (https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) `/var` should be used for changing files.

I would suggest using `/opt` and ideally a subdir in there - e.g. `/opt/mnist/`

WDYT?

andreyvelich replied:
Good catch. What do you think about `/app`?

vpavlin replied:
Any particular reason you do not want to go with existing directory structure in `/`? I don't think it is a good practice in linux systems to put code in "random" dirs in `/`, but feel free to convince me otherwise:)

andreyvelich replied:
Do you mean in `/examples/v1alpha3/metrics-with-timestamp` ?

vpavlin replied:
> Do you mean in `/examples/v1alpha3/metrics-with-timestamp` ?

Is this answer to @johnugeorge'S question?:-)

andreyvelich replied:
> Any particular reason you do not want to go with existing directory structure in `/`? I don't think it is a good practice in linux systems to put code in "random" dirs in `/`, but feel free to convince me otherwise:)

To this question :) 
Which `WORKDIR` you think will be better?

vpavlin replied:
Ah, sorry:) as I suggested in my first comment - I would go for `/opt/mnist/mnist.py` or to make it extremely obvious what it is (although that is already captured in the image name, but you never know..) `/opt/metrics-with-timestamp/mnist.py` or something along those lines

andreyvelich replied:
@johnugeorge @hougangliu What do you think about `WORKDIR` for the example as `/opt/metrics-with-timestamp/mnist.py`

andreyvelich commented:
I changed example from pytorch to mxnet.
I think that would be more useful for the Katib user.
Also, I use kubeflowkatib docker repository now.
/cc @johnugeorge @vpavlin 
/hold cancel

andreyvelich commented:
I changed `mxnet-mnist-timestamp` image to `mxnet-mnist` as we discussed here: https://github.com/kubeflow/katib/pull/1008#discussion_r364820947. 
I think it will be better to name it `mxnet-mnist` than `mxnet-mnist-example`. 
In case that some users use old yaml files for their Katib experiments with `mxnet-mnist-example` image and, if we push new image to `mxnet-mnist-example`, it will brake current Katib experiments.

I renamed images in according yaml files, please take a look.

/cc @johnugeorge @hougangliu @gaocegege

andreyvelich commented:
I changed CI cluster `num nodes` to 8.
It decreased CI test time ~ 30%.
What do you think about this change @johnugeorge @hougangliu @gaocegege ?

johnugeorge commented on test/scripts/v1alpha3/create-cluster.sh:
```diff
@@ -33,6 +33,9 @@ echo "Creating GPU cluster"
 gcloud --project ${PROJECT} beta container clusters create ${CLUSTER_NAME} \
     --zone ${ZONE} \
     --machine-type=n1-standard-8 \
+    --num-nodes=8 \
```
This should not be changed since overall resources are fixed

johnugeorge commented:
/lgtm

johnugeorge commented:
/approve
